### PR TITLE
[Enhancement] Use multi phase aggregate as count distinct default plan

### LINKED ISSUE #5115: Single count distinct is better to use 4 stage aggregate when the number of BEs is large
### Environment
- 4 BEs, each of which has 16 vCPUs.
- 100g SSB.
- query
```sql
SELECT COUNT (DISTINCT LO_ORDERKEY) FROM lineorder_flat;
```

### Test Result
- If it uses 4 stage aggregate, it costs 930ms.
- If it uses 2 stage aggregate, it costs 7540ms.

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #5115 #5725 

## Problem Summary(Required) ：

Because of the StarRocks Pipline engine, the multi-phase aggregate is faster than before, we can use it as count distinct default plan.

### CODE REVIEW

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java [hunk 1]
```diff
@@ -186,78 +186,6 @@ public boolean isDistinctAggFun(CallOperator aggOperator, PhysicalHashAggregateO
                     aggOperator.isDistinct();
         }
 
-        // some agg function has extra cost, we need compute here
-        // eg. MULTI_DISTINCT_COUNT function needs compute extra memory cost
```
[the remark this answers is not in the document]
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java:190]: No, multi count distinct function in SQL can only use MULTI_DISTINCT_COUNT function. like select count(disitnct a), count(dstinct b) from table

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 5 / 7 (71.43%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/SplitAggregateRule.java|4|6|66.67%|
|:large_blue_circle:|com/starrocks/sql/optimizer/cost/CostModel.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java:190]: So why remove the extra cost compute?

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**